### PR TITLE
Fixed issues with flag processing in the C++/C translations

### PR DESCRIPTION
**Pull request by sgnn7**
Old flags were using the logic of (flags & FLAG_FOO != 0) which doesn't
produce correct reuslts if they are not additive so now we use
((flags & FLAG_FOO) == FLAG_FOO) to work around this issue.

### CODE REVIEW

**Comment by sgnn7** [comment]
https://phabricator.endlessm.com/T17251

File: xapian-glib/xapian-database.cc [hunk 1]
```diff
@@ -638,11 +663,11 @@ xapian_database_compact_to_path (XapianDatabase             *self,
 
   int real_flags = 0;
 
-  if (flags & XAPIAN_DB_COMPACT_NO_RENUMBER)
+  if ((flags & XAPIAN_DB_COMPACT_NO_RENUMBER) == XAPIAN_DB_COMPACT_NO_RENUMBER)
```
Comment (ebassi) [inline xapian-glib/xapian-database.cc]: This is not right; `Xapian::DBCOMPACT_*` values **are** a bitmask, and they are handled appropriately in the Xapian source code.

File: xapian-glib/xapian-database.cc [hunk 2]
```diff
@@ -672,11 +697,11 @@ xapian_database_compact_to_fd (XapianDatabase             *self,
...
```
Comment (ebassi) [inline xapian-glib/xapian-database.cc]: Same as above: database compaction flags are really flags.

**Comment by sgnn7** [comment]
@ebassi Some of the new content is a bit above my head in regards to Xapian internals but LGTM.

**Pull request merged**